includes/components/class-router.php

```diff
@@ -681,4 +681,15 @@ public function disable_page_redirect( $url ) {
 
 		return $url;
 	}
+
+	/**
+	 * Redirect to Not Found page.
+	 */
+	public function redirect_not_found_page() {
```

Comment:
Please consult before adding new helpers, without docs we'll simply forget about them. In this case, if it's the first occurrence of /404 then using this is ok:

return home_url( '/404' )

Remember that redirect functions of routes should return something (false for no redirect, true for redirect to the next item for chained routes and home page for regular ones, and some URL for specific redirect). Calling wp_redirect is not needed here, redirect function handler already calls it.

Reply:
You can add @todo before returning to improve this later, e.g. "@todo: add a custom route or helper for 404".

Reply:
Please name branches based on the task name, in this case "fix-404-for-user-profiles" or "-for-users" would be ok, but in the current branch only the "user" word is common with the task title - was a bit hard to find it.

Reply:
It seems that syncing with master causes conflicts, resolving them or adding a new branch may be required.